pkg/controller/worker/genericactuator/actuator_resore.go

```diff
@@ -0,0 +1,172 @@
+	// Make a deep copy to ensure that no one will change the state during the restoration
+	workerCopy := worker.DeepCopy()
+
+	if workerCopy.Status.State == nil || len(workerCopy.Status.State.Raw) <= 0 {
+		return nil
+	}
+
+	// Parse the worker state to MachineDeploymentStates
+	workerState := make(map[string]*workercontroller.MachineDeploymentState)
+	if err := json.Unmarshal(workerCopy.Status.State.Raw, &workerState); err != nil {
+		return err
+	}
+
+	// Attach the parsed MachineDeploymentStates to the wanted MachineDeployments
+	for index, wantedMachineDeployment := range wantedMachineDeployments {
+		wantedMachineDeployments[index].State = workerState[wantedMachineDeployment.Name]
+	}
+
+	return nil
+}
+
+func (a *genericActuator) deployMachineSetsAndMachines(ctx context.Context, worker *extensionsv1alpha1.Worker, wantedMachineDeployments workercontroller.MachineDeployments) error {
+	for _, wantedMachineDeployment := range wantedMachineDeployments {
+		if wantedMachineDeployment.State == nil || wantedMachineDeployment.State.MachineSet == nil {
+			continue
+		}
+
+		// Extract the MachineSet
+		machineSet := &machinev1alpha1.MachineSet{}
+		if _, _, err := a.decoder.Decode(wantedMachineDeployment.State.MachineSet.Raw, nil, machineSet); err != nil {
+			return err
+		}
+
+		// Create the MachineSet if not already exists. We do not care about the MachineSet status
+		// because the MCM will update it
+		if err := a.client.Create(ctx, machineSet); err != nil && !apierrors.IsAlreadyExists(err) {
+			return err
+		}
+
+		// Deploy each machine owned by the MachineSet which was restored above
+		for _, rawMachine := range wantedMachineDeployment.State.Machines {
+			if rawMachine.Raw == nil {
+				continue
+			}
+			// Unmarshal the Machine
+			machine := &machinev1alpha1.Machine{}
+			if _, _, err := a.decoder.Decode(rawMachine.Raw, nil, machine); err != nil {
+				return err
+			}
+
+			// Create the machine if it not exists already
+			err := a.client.Create(ctx, machine)
+			if err != nil && !apierrors.IsAlreadyExists(err) {
+				return err
+			}
+
+			// Attach the Shoot node to the Machine status
+			node := machine.Status.Node
+			if err := a.updateStatus(ctx, machine, apierrors.IsAlreadyExists(err), func() error {
+				machine.Status.Node = node
+				return nil
+			}); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// alreadyExists must be returned from from apierrors.IsAlreadyExists(err) where err commes from client.Create.
+// This function is needed because wen you call client.Create and it successed if you immediately call TryUpdateStatus
+// it will return an error stating that the Object does not exists.
+// This bug comes form the fact that exponentialBackoff in TryUpdateStatus does not retry if client.Get returns error
+// so the function exits immediately with error
```

Comment:
@rfranzke should we update TrypUpdateStatus to respect the exponentialBackoff when the initial `Get` returns an error (I'd imagine that that would be the desired behaviour actually)